src/main/java/io/cryostat/net/web/http/api/beta/MatchExpressionsPostHandler.java

```diff
@@ -0,0 +1,183 @@
+        return EnumSet.of(ResourceAction.CREATE_MATCH_EXPRESSION);
+    }
+
+    @Override
+    public String path() {
+        return basePath() + PATH;
+    }
+
+    @Override
+    public List<HttpMimeType> produces() {
+        return List.of(HttpMimeType.JSON);
+    }
+
+    @Override
+    public List<HttpMimeType> consumes() {
+        return List.of(HttpMimeType.MULTIPART_FORM, HttpMimeType.URLENCODED_FORM);
+    }
+
+    @Override
+    public boolean isAsync() {
+        return false;
+    }
+
+    @Override
+    public boolean isOrdered() {
+        return true;
+    }
+
+    @Override
+    public IntermediateResponse<MatchedMatchExpression> handle(RequestParameters params)
+            throws ApiException {
+        String matchExpression = params.getFormAttributes().get("matchExpression");
+        String targets = params.getFormAttributes().get("targets");
+        if (StringUtils.isBlank(matchExpression)) {
+            throw new ApiException(400, "'matchExpression' is required.");
+        }
+        try {
+            if (StringUtils.isNotBlank(targets)) {
+                Set<ServiceRef> matched;
+                List<ServiceRef> parsedTargets = this.expressionManager.parseTargets(targets);
+                matched =
+                        this.expressionManager.resolveMatchingTargets(
+                                matchExpression, parsedTargets);
+
+                return new IntermediateResponse<MatchedMatchExpression>()
+                        .statusCode(200)
+                        .body(new MatchedMatchExpression(matchExpression, matched));
+            } else {
+                int id = this.expressionManager.addMatchExpression(matchExpression);
+                Optional<MatchExpression> opt = this.expressionManager.get(id);
+                if (opt.isEmpty()) {
+                    throw new ApiException(500, "Failed to add match expression");
+                }
+                MatchExpression expr = opt.get();
+                notificationFactory
+                        .createBuilder()
+                        .metaCategory("MatchExpressionAdded")
+                        .metaType(HttpMimeType.JSON)
+                        .message(Map.of("id", id, "matchExpression", expr.getMatchExpression()))
+                        .build()
+                        .send();
+                return new IntermediateResponse<MatchedMatchExpression>()
+                        .statusCode(201)
+                        .addHeader(HttpHeaders.LOCATION, String.format("%s/%d", path(), id))
+                        .body(new MatchedMatchExpression(expr));
+            }
+        } catch (RollbackException e) {
+            if (ExceptionUtils.indexOfType(e, ConstraintViolationException.class) >= 0) {
+                throw new ApiException(400, "Duplicate matchExpression", e);
+            }
+            throw new ApiException(500, e);
+        } catch (MatchExpressionValidationException e) {
```

Comment:
Should we also catch any json parsing exception? For example,

```bash
 $ https --verify=no --auth=user:pass -f POST :8181/api/beta/matchExpressions matchExpression="target.alias == 'es.andrewazor.demo.Main'" targets="[{aasd}]"
HTTP/1.1 500 Internal Server Error
content-encoding: gzip
content-length: 182
content-type: application/json

{
    "data": {
        "reason": "com.google.gson.JsonSyntaxException: com.google.gson.stream.MalformedJsonException: Expected ':' at line 1 column 8 path $[0].aasd"
    },
    "meta": {
        "status": "Internal Server Error",
        "type": "text/plain"
    }
}
```

This should be `400`?